### PR TITLE
Added document.baseURI check for can-util/js/base-url

### PR DESCRIPTION
**Pull request by sarathnair**
for #223 

Added code to check if document.baseURI exists. If it exists return the same

### CODE REVIEW

**Comment by sarathnair** [comment]
I have a doubt though, for the current page document.baseURI returns `https://github.com/canjs/can-util/pull/227`. Is this expected with can-util/js/base-url? I saw that location.href is manipulated to trim off the last path in URL so that it returns `https://github.com/canjs/can-util/pull`. If this is what expected I can make corresponding changes to this commit.

File: js/base-url/base-url-test.js [hunk 1]
```diff
@@ -7,7 +7,9 @@ QUnit.module("can-util/js/base-url");
 test("basics", function(){
 	var global = getGlobal();
 
-	if(global.location) {
+	if (global.document && 'baseURI' in global.document) {
```
Comment (matthewp) [inline js/base-url/base-url-test.js]: use can-util/dom/document/document instead of `global.document`.

**Comment by matthewp** [comment]
Just one comment, otherwise looks good!

**Comment by sarathnair** [comment]
Done! Made the changes mentioned.

**Comment by matthewp** [comment]
Cool, thanks.

**Pull request merged**